lua/nvim-ts-autotag/utils.lua

```diff
@@ -2,6 +2,68 @@ local log = require("nvim-ts-autotag._log")
 local get_node_text = vim.treesitter.get_node_text
 local M = {}
 
+---@return boolean
+function M.is_react_file()
+    local ft = vim.bo.ft
+    -- check filetypes first.
+    if ft == "javascriptreact" or ft == "typescriptreact" then
+        return true
+    elseif ft ~= "javascript" then
+        return false
+    end
+
```

Comment:
Just a simple comment to add, as the logic around the checks made me scratch my head for a sec before I realized *why* you were doing that. (Do correct the suggestion if I'm wrong).

```suggestion
-- If we are in a `javascript` file, then check the content to see if the
-- current file counts as a react file
```

Reply:
That's precisely the problem that function is trying to solve, you nailed it. The low-hanging-fruit is the case where we are in a `jsx` or `tsx` file, or not in a `js` file. We need logic to determine if react is being used if we are in a `js` file. Javascript. I love it. lol